### ISSUE TITLE
Failed to load plugin 'no-date-parsing' declared in

### ISSUE THREAD

**Issue by binary64** [amzn/eslint-plugin-no-date-parsing#4]
I have `"eslint-plugin-no-date-parsing": "0.2.0",` installed in my package.json and `'no-date-parsing',` in my eslintrc plugins, and `'no-date-parsing/no-date-parsing': 'error',` in my overrides>*.ts>rules, but I get this error:
```
> nx run blah:lint
Linting "blah"...
Failed to load plugin 'no-date-parsing' declared in 'libs/blah/.eslintrc.js » ../../.eslintrc.js': Class extends value undefined is not a constructor or null
```

**Comment by joeframbach** [comment]
Hello. Could you share your package.json dev-dependencies? Perhaps there is an issue with a newer or older version of eslint that I am not testing.

**Comment by Merott** [comment]
Running into the same error:

```
[Error - 11:38:20 AM] TypeError: Failed to load plugin 'no-date-parsing' declared in '.eslintrc.js': Class extends value undefined is not a constructor or null
    at Object.<anonymous> (/<redacted>/node_modules/@typescript-eslint/experimental-utils/dist/ts-eslint/CLIEngine.js:12:34)
    at Module._compile (node:internal/modules/cjs/loader:1165:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
    at Module.load (node:internal/modules/cjs/loader:1035:32)
    at Module._load (node:internal/modules/cjs/loader:876:12)
    at c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1059:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/<redacted>/node_modules/@typescript-eslint/experimental-utils/dist/ts-eslint/index.js:14:14)
    at Module._compile (node:internal/modules/cjs/loader:1165:14)
```

```json
"devDependencies": {
  "@axe-core/react": "^4.4.3",
  "@babel/core": "^7.18.13",
  "@graphql-codegen/cli": "^2.9.1",
  "@graphql-codegen/typescript": "^2.7.2",
  "@graphql-codegen/typescript-operations": "^2.5.2",
  "@graphql-codegen/typescript-resolvers": "^2.7.2",
  "@storybook/addon-actions": "^6.5.10",
  "@storybook/addon-essentials": "^6.5.10",
  "@storybook/addon-interactions": "^6.5.10",
  "@storybook/addon-links": "^6.5.10",
  "@storybook/builder-webpack5": "^6.5.10",
  "@storybook/manager-webpack5": "^6.5.10",
  "@storybook/react": "^6.5.10",
  "@storybook/testing-library": "^0.0.13",
  "@testing-library/dom": "^8.16.0",
  "@testing-library/jest-dom": "^5.16.4",
  "@testing-library/react": "^13.3.0",
  "@testing-library/user-event": "^14.3.0",
  "@types/cryptr": "^4.0.1",
  "@types/eslint": "^8.4.5",
  "@types/micro-cors": "^0.1.2",
  "@types/node": "^18.0.6",
  "@types/nodemailer": "^6.4.6",
  "@types/prettier": "^2.6.3",
  "@types/react": "^18.0.15",
  "@types/react-dom": "^18.0.6",
  "@typescript-eslint/eslint-plugin": "^5.30.6",
  "@typescript-eslint/parser": "^5.30.6",
  "@vitejs/plugin-react": "^2.0.0",
  "autoprefixer": "^10.4.8",
  "babel-loader": "^8.2.5",
  "chokidar": "^3.5.3",
  "chokidar-cli": "^3.0.0",
  "clipboardy": "^3.0.0",
  "cryptr": "^6.0.3",
  "dotenv": "^16.0.1",
  "esbuild": "^0.14.49",
  "eslint": "8.19.0",
  "eslint-config-next": "^12.2.2",
  "eslint-config-prettier": "^8.5.0",
  "eslint-import-resolver-typescript": "^3.3.0",
  "eslint-plugin-import": "^2.26.0",
  "eslint-plugin-no-date-parsing": "^0.2.0",
  "eslint-plugin-storybook": "^0.6.4",
  "happy-dom": "^6.0.4",
  "husky": "^8.0.1",
  "npm-run-all": "^4.1.5",
  "postcss": "^8.4.16",
  "prettier": "^2.7.1",
  "prettier-plugin-tailwindcss": "^0.1.12",
  "pretty-quick": "^3.1.3",
  "tailwind-scrollbar": "^1.3.2",
  "tailwindcss": "^3.1.8",
  "ts-node": "^10.9.1",
  "typescript": "^4.7.4",
  "vite-tsconfig-paths": "^3.5.0",
  "vitest": "^0.18.1"
}
```

**Comment by JonKrone** [comment]
Hey there! I am really excited to add this lint rule as we're constantly fighting against the natural instinct to use the builtin Date for parsing date strings, but we're seeing this same plugin import issue.

It seems to be related to eslint v8 as similar github Issues suggest ([tslint here](https://github.com/eslint/eslint/issues/15149)). I have not looked too far into it, but [it may be related to ESM compatability](https://stackoverflow.com/a/70273994/8222788).

I don't maintain any packages and am not super familiar with what changes might be needed, but I remember [this post by Sindresorhus](https://github.com/sindresorhus/meta/discussions/15) was a pragmatic read on how they're migrating their hundreds of packages.

<details><summary>Our package.json</summary>
```json
"devDependencies": {
    "@next/eslint-plugin-next": "^12.3.0",
    "@typescript-eslint/eslint-plugin": "^5.36.2",
    "@typescript-eslint/parser": "^5.36.2",
    "eslint": "^8.23.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "27.0.2",
    "eslint-plugin-jest-playwright": "^0.9.0",
    "eslint-plugin-no-date-parsing": "0.2.0",
    "eslint-plugin-react": "^7.31.7",
    "eslint-plugin-simple-import-sort": "^8.0.0",
  },
```
</details>

**Comment by ElliotAtAcuityMD** [comment]
Running into this as well. Has anyone figured out how to get this working?
```
    "@typescript-eslint/eslint-plugin": "^5.43.0",
    "@typescript-eslint/parser": "^5.32.0",
     "eslint-plugin-no-date-parsing": "^0.2.0",
     "eslint": "^8.21.0",
     "typescript": "^4.6.2",
```

**Comment by joeframbach** [comment]
I have filed a PR, please review.

**Comment by joeframbach** [comment]
I have published v1.0.0 which supports eslint v8 with typescript-eslint v5.
